SingularityUI/app/utils.es6

```diff
@@ -486,6 +487,32 @@ const Utils = {
     }
     const authToken = JSON.parse(authCookie)[config.authTokenKey];
     return `Bearer ${ authToken }`;
+  },
+
+  template(template, data) {
+    const start = "{{";
+    const end = "}}";
+    const path = "[a-z0-9_$][\\.a-z0-9_]*";
+    const pattern = new RegExp(start + "\\s*("+ path +")\\s*" + end, "gi");
+    try {
+      return template.replace(pattern, (tag, token) => {
+        const tokenPath = token.split(".");
+        let value = data;
+        let i = 0;
+
+        for (; i < tokenPath.length; i++){
+          value = value[tokenPath[i]];
+          if (value === undefined){
+            throw tokenPath[i] + "' not found in " + tag;
+          }
+          if (i === tokenPath.length - 1){
+            return value;
+          }
+        }
+      });
+    } catch (err) {
+      return undefined;
```

Comment:
Is there a reason you're returning `undefined` rather than `null`?

Reply:
no particular reason. Is there a preference/reasoning for one over the other?